acceptance/tests/restart_host_run_puppet.rb

```diff
@@ -22,8 +22,14 @@
   step "restart each agent" do
     applicable_agents.each do |agent|
       agent.reboot
+      # BKR-812
+      sleep 1 if !agent.up?
       assert(agent.up?, "Agent #{agent} should be up after reboot")
-      on(agent, puppet('resource service pxp-agent ')) do |result|
+      opts = {
+        :max_retries => 10,
+        :retry_interval => 1,
+      }
+      retry_on(agent, puppet('resource service pxp-agent '), opts) do |result|
```

Comment:
interestingly this test fails locally with the current stable PA build (12841f398f1ea187fccda2a0ed99cd505df7de2a) without this retry_on, and works with it.  i'm not quite sure why.